File: Beecrowd/1221 Primo Rápido/Main.java
import java.io.IOException;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);

        int n, x;

        n = sc.nextInt();

        for (int i = 0; i < n; i++) {
            x = sc.nextInt();

            if (EPrimo(x))
                System.out.println("Prime");
            else
                System.out.println("Not Prime");
        }

        sc.close();
    }

    public static boolean EPrimo(int x) {
        int i;
        double raiz;
        if (x == 2)
            return true;
        else if (x % 2 == 0)
            return false;
        else {
            i = 3;
            raiz = Math.sqrt(x);
            while (i <= raiz) {
                if (x % i == 0)
                    return false;
                i = i + 2;
            }
        }
        return true;
    }

}

File: Beecrowd/1221 Primo Rápido/main.py
import math

def EPrimo(pX):
  if pX % 2 == 0 and pX > 2:
    return False
  else:
    raiz = math.sqrt(pX)
    for i in range(3, int(raiz + 1), 2):
      if pX % i == 0:
        return False
  return 1
	
N = int(input())

for i in range(0, N):
  X = int(input())
  if EPrimo(X):
    print("Prime")
  else:
    print("Not Prime")